### ISSUE TITLE
How to use API for FullNode

### ISSUE THREAD

**Issue by akme** [ton-blockchain/ton#76]
How can we use [ton_api.tl](https://github.com/ton-blockchain/ton/blob/master/tl/generate/scheme/ton_api.tl) (not from C++, e.g. Golang) for accessing Full Node as `validator-engine-console` do? Should we still use tonlibclientjson or there is any other lib we can use?

**Comment by ton-blockchain** [comment]
The API available to validator-engine-console is only a small part of ton_api.tl (its last ---types--- and ---functions--- section). We do not intend to provide external bindings to the client implementation of this API. You can either use validator-engine-console from suitable scripts or using pipes to redirect stdin/stdout; or make a copy of validator-engine-console.cpp and add whatever extra functions or bindings you need.

**Comment by akme** [comment]
Sad, but got it.. and what about lite_api? same story as ton_api?

**Comment by ton-blockchain** [comment]
lite_api.tl is the LiteServer API used by LiteClient, Blockchain Explorer and TONLib.
The idea behind TONLib is that it (remotely) uses the LiteServer API and (locally) provides a more high-level TONLib API, to be used in client applications. This API can be used from a variety of programming languages (if somebody provides proper bindings).

Using the bare LiteServer API from programming languages other than C++ would be not so easy, because the fields in LiteServer queries have to be serialized or deserialized and checked using methods provided in TON Block (C++) library and related libraries. Another problem is that an application directly using LiteServer API would also have to use the network and actor machinery of the TON Library (essentially the "main loop" of all TON programs), which may be unsuitable for interoperation with other languages. Perhaps integrating a scripting language (lua or fift) into (an extended version of) the LiteClient is a better solution, but we do not plan to do this any time soon.

In short, if you think about incorporating the LiteServer API into a library to be used from another programming language, or from a C++ program not using TON actors and network, you probably actually want to use TONLib instead, maybe with some extra features.

**Comment by akme** [comment]
Thank you for explanation!
